### PR TITLE
Release v0.6.0

### PR DESCRIPTION
**Pull request by z4y4ts**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,19 +7,18 @@ python:
 addons:
   apt:
     packages:
-    - ruby1.9.3
```
Comment (auraz) [inline .travis.yml]: Like it :)

File: video_xblock/static/js/studio-edit.js [hunk 2]
```diff
@@ -357,6 +359,63 @@ function StudioEditableXBlock(runtime, element) {
...
+            var success_message = response['success_message'];
+            if (success_message) {
+                if (response.transcripts) {
+                    response.transcripts.forEach(function transcript(item) {
+                        includeLang = transcriptsValue.find(
+                            function existLanguage(element) {
+                                return element.lang == item.lang;
+                            }
+                        );
+                        if (!includeLang) {
```
Comment (auraz) [inline video_xblock/static/js/studio-edit.js]: please add comment why it is needed

File: video_xblock/static/js/studio-edit.js [hunk 3]
```diff
@@ -357,6 +359,63 @@ function StudioEditableXBlock(runtime, element) {
...
+            else if (error_message) {
+                message = error_message;
+                status = ERROR;
+            }
+        })
+        .fail(function(jqXHR) {
+            message = gettext('This may be happening because of an error with our server or your ' +
+                'internet connection. Try refreshing the page or making sure you are online.');
+            status = ERROR;
+            if (jqXHR.responseText) { // Is there a more specific error message we can show?
```
Comment (auraz) [inline video_xblock/static/js/studio-edit.js]: address the question please

**Comment by auraz** [comment]
Please add JS tests.

File: video_xblock/static/html/studio-edit-playmedia_apikey.html [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{% load i18n %}
```
Comment (polesye) [inline video_xblock/static/html/studio-edit-playmedia_apikey.html]: why do you mix underscores and dashes in file names?

File: video_xblock/static/js/studio-edit.js [hunk 5]
```diff
@@ -357,6 +359,63 @@ function StudioEditableXBlock(runtime, element) {
...
+                    });
+                }
+                message = success_message;
+                status = SUCCESS;
+            }
+            else if (error_message) {
+                message = error_message;
+                status = ERROR;
+            }
+        })
```
Comment (polesye) [inline video_xblock/static/js/studio-edit.js]: https://refactoring.guru/replace-nested-conditional-with-guard-clauses

File: video_xblock/static/js/studio-edit.js [hunk 6]
```diff
@@ -357,6 +359,63 @@ function StudioEditableXBlock(runtime, element) {
...
+            message = gettext('This may be happening because of an error with our server or your ' +
+                'internet connection. Try refreshing the page or making sure you are online.');
+            status = ERROR;
+            if (jqXHR.responseText) { // Is there a more specific error message we can show?
+                message = extractErrorMessage(jqXHR.responseText);
+            }
+            runtime.notify('error', {title: gettext('Unable to update settings'), message: message});
+        })
+        .always(function() {
+            showStatus($('.threeplaymedia.status'), status, message);
```
Comment (polesye) [inline video_xblock/static/js/studio-edit.js]: Can we simply extract message and status here? Why do we use external variables?

File: video_xblock/static/js/studio-edit.js [hunk 7]
```diff
@@ -579,6 +638,14 @@ function StudioEditableXBlock(runtime, element) {
         authenticateVideoApi($data);
     });
 
+    $3playmediaTranscriptsApi.on('click', function(event) {
+        event.preventDefault();
+        event.stopPropagation();
+        var $api_key = $('.threeplaymedia-api-key', element).val();
```
Comment (polesye) [inline video_xblock/static/js/studio-edit.js]: @z4y4ts is it secure to use this key in client?

File: video_xblock/static/js/studio-edit.js (hunk 6 above)
Comment (z4y4ts) [inline video_xblock/static/js/studio-edit.js]: @polesye I'm not sure I'm getting you here. Can you please elaborate?

File: video_xblock/static/js/studio-edit.js (hunk 7 above)
Comment (z4y4ts) [inline video_xblock/static/js/studio-edit.js]: In this case yes. We're getting this API key from the user in the first place. And the key does not leave the same view where user enters it.

File: video_xblock/static/html/studio-edit-playmedia_apikey.html (hunk 4 above)
Comment (z4y4ts) [inline video_xblock/static/html/studio-edit-playmedia_apikey.html]: @dorosh please address this.

File: video_xblock/static/js/studio-edit.js (hunk 2 above)
Comment (z4y4ts) [inline video_xblock/static/js/studio-edit.js]: @dorosh please address this.

File: video_xblock/static/js/studio-edit.js (hunk 5 above)
Comment (z4y4ts) [inline video_xblock/static/js/studio-edit.js]: @dorosh please address this.

File: video_xblock/static/js/studio-edit.js (hunk 3 above)
Comment (z4y4ts) [inline video_xblock/static/js/studio-edit.js]: Well, yeah that's misleading indeed. The question is actually being addressed on the next line.

@dorosh Please change it to `Try to get more specific error message we can show to user`

File: video_xblock/static/js/studio-edit.js (hunk 6 above)
Comment (polesye) [inline video_xblock/static/js/studio-edit.js]: nevermind

**Comment by bryanlandia** [comment]
Sorry this isn't the best place to make these comments but I don't have access to Issues, and the files were not part of the PR.  


* `README.md `  https://github.com/raccoongang/xblock-video/blob/dev/README.md#installation
Refers to `make deps` and  `deps` is not a valid target in the `Makefile`.  

```
edxapp@precise64:~/venvs/edxapp/src/video-xblock$ make deps
make: *** No rule to make target `deps'.  Stop.
```
It is also missing a `bower install` to get the standard (non-test) dependencies.  Am I missing something?

* There seems to be a version incompatibility issue with the dev dependencies in `package.json`. 
```
edxapp@precise64:~/venvs/edxapp/src/video-xblock$ npm install
npm WARN package.json @ No repository field.
npm WARN engine eslint@3.18.0: wanted: {"node":">=4"} (current: {"node":"0.10.37","npm":"1.4.28"})
npm WARN engine shelljs@0.7.7: wanted: {"node":">=0.11.0","iojs":"*"} (current: {"node":"0.10.37","npm":"1.4.28"})
npm WARN engine strip-bom@3.0.0: wanted: {"node":">=4"} (current: {"node":"0.10.37","npm":"1.4.28"})
npm WARN engine esprima@3.1.3: wanted: {"node":">=4"} (current: {"node":"0.10.37","npm":"1.4.28"})
npm WARN engine string-width@2.0.0: wanted: {"node":">=4"} (current: {"node":"0.10.37","npm":"1.4.28"})
npm WARN engine is-fullwidth-code-point@2.0.0: wanted: {"node":">=4"} (current: {"node":"0.10.37","npm":"1.4.28"})
npm WARN engine co@4.6.0: wanted: {"iojs":">= 1.0.0","node":">= 0.12.0"} (current: {"node":"0.10.37","npm":"1.4.28"})
npm WARN engine string-width@2.0.0: wanted: {"node":">=4"} (current: {"node":"0.10.37","npm":"1.4.28"})
npm WARN engine esprima@3.1.3: wanted: {"node":">=4"} (current: {"node":"0.10.37","npm":"1.4.28"})
npm WARN engine is-fullwidth-code-point@2.0.0: wanted: {"node":">=4"} (current: {"node":"0.10.37","npm":"1.4.28"})
npm WARN engine co@4.6.0: wanted: {"iojs":">= 1.0.0","node":">= 0.12.0"} (current: {"node":"0.10.37","npm":"1.4.28"})
npm ERR! peerinvalid The package eslint does not satisfy its siblings' peerDependencies requirements!
npm ERR! peerinvalid Peer eslint-config-airbnb-base@3.0.1 wants eslint@^2.9.0
npm ERR! peerinvalid Peer eslint-plugin-dollar-sign@1.0.0 wants eslint@>=1.0.0
npm ERR! peerinvalid Peer eslint-plugin-import@1.16.0 wants eslint@2.x - 3.x

npm ERR! System Linux 3.2.0-23-generic
npm ERR! command "/usr/bin/node" "/usr/bin/npm" "install"
npm ERR! cwd /edx/app/edxapp/venvs/edxapp/src/video-xblock
npm ERR! node -v v0.10.37
npm ERR! npm -v 1.4.28
npm ERR! code EPEERINVALID
npm ERR! 
npm ERR! Additional logging details can be found in:
npm ERR!     /edx/app/edxapp/venvs/edxapp/src/video-xblock/npm-debug.log
npm ERR! not ok code 0
```

**Comment by z4y4ts** [comment]
Hi @bryanlandia 

Sorry for the mess, I guess I've accidently disabled issues here.
Anyways, I'd like to have some feedback from you to be sure documentation is actually works for you. So I've created [an issue and replied there](https://github.com/raccoongang/xblock-video/issues/173).

**Pull request merged**